src/memray/_memray/tracking_api.cpp

```diff
@@ -290,6 +328,7 @@ Tracker::Tracker(
 
     d_background_thread = std::make_unique<BackgroundThread>(d_writer, memory_interval);
     d_background_thread->start();
+    d_sampler.setSamplingInterval(sampling_interval);
```

Comment:
We're only calling this in the thread where the `Tracker` is created. Unless I'm missing something, other threads are winding up with 0 for all the member variables of `d_sampler`.

I think what we could do is call `setSamplingInterval` unconditionally in `trackAllocationImpl` and `trackDeallocationImpl`, and update `setSamplingInterval` to short-circuit in the case where the old sampling interval and the new one are the same.